oracle/oracle.go

```diff
@@ -199,8 +199,10 @@ func (o *OracleImpl) fetchPrices(provider providertypes.Provider[oracletypes.Cur
 
 	timeFilteredPrices := make(map[oracletypes.CurrencyPair]*big.Int)
 	for pair, result := range prices {
+		floatValue, _ := result.Value.Float64() // we ignore the accuracy in this conversion
+
 		// update price metric
-		o.metrics.UpdatePrice(provider.Name(), string(provider.Type()), pair.String(), float64(result.Value.Int64()))
+		o.metrics.UpdatePrice(provider.Name(), string(provider.Type()), pair.String(), floatValue)
```

Comment:
context: I make this change as the `.Int64()` method on big.Int's behavior is undefined for non-int64 values, hence we can report incorrect information in the common-case